cmd/swarmd/main.go

```diff
@@ -126,6 +128,12 @@ var (
 				return err
 			}
 
+			resources := &api.Resources{ScalarResources: map[string]float64{}}
+			err = flagparser.ParseUserDefinedResources(cmd.Flags(), resources, "resources", false)
```

Comment:
I think it should follow other cmd.Flags() to read parameters. Validation should be done separately, like other flags.